scan/vulnerability.go

```diff
@@ -0,0 +1,56 @@
+package scan
+
+import (
+	"fmt"
+	"log"
+	"net/url"
+
+	"github.com/FiloSottile/Heartbleed/heartbleed"
+)
+
+// Vulnerability contains scanners to detect vulnerabilities such as Heartbleed
+var Vulnerability = &Family{
+	Description: "Scans a host to detect TLS vulnerabilities.",
+	Scanners: map[string]*Scanner{
+		"Heartbleed": {
+			"Scan for Heartbleed vulnerability",
+			heartbleedScan,
+		},
+	},
+}
+
+func heartbleedScan(addr, hostname string) (grade Grade, output Output, err error) {
+	tgt := &heartbleed.Target{
+		Service: "https",
+		HostIp:  hostname,
+	}
+
+	u, err := url.Parse(tgt.HostIp)
+	if err == nil && u.Host != "" {
+		tgt.HostIp = u.Host
+		if u.Scheme != "" {
+			tgt.Service = u.Scheme
+		}
+	}
+
+	out, err := heartbleed.Heartbleed(tgt,
+		[]byte("github.com/FiloSottile/Heartbleed"), true)
+	if err == heartbleed.Safe {
+		str := fmt.Sprintf("%v - SAFE", tgt.HostIp)
```

Comment:
There's no need to repeat the `hostname` in the outputs. I don't think I would set any outputs for any of these (the relevant messages are already set in the error).